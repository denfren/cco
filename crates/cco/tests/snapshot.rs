//! Snapshot tests
//!
//! Loads each *.hcl file in /tests/ individually and compares if the
//! output of expression `test` changes.

#[test]
fn snapshots() {
    tracing_subscriber::fmt()
        .with_env_filter(tracing_subscriber::EnvFilter::from_env("CCO_LOG"))
        .with_writer(std::io::stderr)
        .init();

    insta::glob!("*.hcl", |path| {
        let mut documents = cco::hcl_documents::HclDocuments::default();
        let reader = std::fs::read_to_string(path).unwrap();
        documents.insert(
            hcl_edit::parser::parse_body(&reader).unwrap(),
            Some(path.to_owned()),
        );
        let documents =
            cco::cco_document::CcoDocument::new(&documents).expect("must be valid cco document");

        let rendered = documents
            .evaluate_in_context(hcl::Variable::unchecked("test").into())
            .expect("valid value");

        insta::assert_yaml_snapshot!(rendered);
    });
}
