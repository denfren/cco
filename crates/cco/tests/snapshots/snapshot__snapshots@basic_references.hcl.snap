---
source: crates/cco/tests/snapshot.rs
expression: rendered
input_file: crates/cco/tests/basic_references.hcl
---
value: test value
ref_to_own_value: test value
ref_to_others_value: example value
attribute_with_object_value:
  value: object value
  abs_ref: test value
ref_into_object: object value

